decentraland/unity-explorer
# fix: Issues with avatar visibility in scenes that use AvatarModifierArea component

## What does this PR change?
Fix #1643 

When we are in a scene that implement the `AvatarModifierArea` component and the scenes around are heavy in terms of performance, when we enter in the modifier areas, sometimes the hiding of the avatar doesn't work or takes too much time (a good example is in the scenes `76,-7`, `76,-6` and `75,-7`).

![image](https://github.com/user-attachments/assets/36c7833c-f1f1-462e-94b6-1883e864fded)

To fix it, I have introduced the checking of `ISceneStateProvider.IsCurrent` in the system `AvatarModifierAreaHandlerSystem` in order to process only when we're inside the current scene.

## How to test the changes?
1. Launch the explorer.
2. Use the command /goto goerli.
3. Navigate to the 'Advanced-Avatar-Swap' scenes at coordinates `76,-7`.
4. Enter the modifier area and check that you avatar is correctly hidden and is replaced by a new model instantaneously.
5. Walk to `75,-7` and check the same.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
